x-pack/test/api_integration/deployment_agnostic/apis/observability/apm/service_overview/dependencies/index.spec.ts

```diff
@@ -0,0 +1,304 @@
+        });
+
+        it('returns two dependencies', () => {
+          expect(response.body.serviceDependencies.length).to.be(2);
+        });
+
+        it('returns opbeans-node as a dependency', () => {
+          const opbeansNode = response.body.serviceDependencies.find(
+            (item) => getName(item.location) === 'opbeans-node'
+          );
+
+          expect(opbeansNode !== undefined).to.be(true);
+
+          const values = {
+            latency: roundNumber(opbeansNode?.currentStats.latency.value),
+            throughput: roundNumber(opbeansNode?.currentStats.throughput.value),
+            errorRate: roundNumber(opbeansNode?.currentStats.errorRate.value),
+            impact: opbeansNode?.currentStats.impact,
+            ...pick(opbeansNode?.location, 'serviceName', 'type', 'agentName', 'environment'),
+          };
+
+          const count = 4;
+          const sum = 21;
+          const errors = 1;
+
+          expect(values).to.eql({
+            agentName: 'nodejs',
+            environment: ENVIRONMENT_NOT_DEFINED.value,
+            serviceName: 'opbeans-node',
+            type: 'service',
+            errorRate: roundNumber(errors / count),
+            latency: roundNumber(sum / count),
+            throughput: roundNumber(count / ((endTime - startTime) / 1000 / 60)),
+            impact: 100,
+          });
+
+          const firstValue = roundNumber(opbeansNode?.currentStats.latency.timeseries[0].y);
+          const lastValue = roundNumber(last(opbeansNode?.currentStats.latency.timeseries)?.y);
+
+          expect(firstValue).to.be(roundNumber(20 / 3));
+          expect(lastValue).to.be(1);
+        });
+
+        it('returns postgres as an external dependency', () => {
+          const postgres = response.body.serviceDependencies.find(
+            (item) => getName(item.location) === 'postgres'
+          );
+
+          expect(postgres !== undefined).to.be(true);
+
+          const values = {
+            latency: roundNumber(postgres?.currentStats.latency.value),
+            throughput: roundNumber(postgres?.currentStats.throughput.value),
+            errorRate: roundNumber(postgres?.currentStats.errorRate.value),
+            impact: postgres?.currentStats.impact,
+            ...pick(postgres?.location, 'spanType', 'spanSubtype', 'dependencyName', 'type'),
+          };
+
+          const count = 1;
+          const sum = 3;
+          const errors = 0;
+
+          expect(values).to.eql({
+            spanType: 'external',
+            spanSubtype: 'http',
+            dependencyName: 'postgres',
+            type: 'dependency',
+            errorRate: roundNumber(errors / count),
+            latency: roundNumber(sum / count),
+            throughput: roundNumber(count / ((endTime - startTime) / 1000 / 60)),
+            impact: 0,
+          });
+        });
+      });
+
+      // UNSUPPORTED TEST CASES - when data is loaded
+      // TODO: These tests should be migrated to use synthtrace
```

Comment:
will it work both against MKI and ESS on Cloud as it is implemented right now?

Reply:
At the moment, the `when data is loaded` cases rely on es archives to load the correct data, which is not supported in the deployment agnostic approach. Those cases are kept in the old test section (to be run with stateful). Adapting these to be compatible would likely require redoing a lot of the cases to work with the synthtrace data so for example, redoing a lot of the snapshots. All the tests that did not use the archives and relied on synthtrace worked fine with the agnostic (MKI/Stateful) approach.

Reply:
thank you for the details, good to know.